README.md

```diff
@@ -1,31 +1,207 @@
+
+
+## Demo code
+
+### **Get list of Terminals**
+
+This simple call is a fast, read only method, that returns a list of all terminals assigned to the store, and their online status.
+
+````python
+ with grpc.secure_channel(target=config.address,
+                             credentials=grpc.ssl_channel_credentials()) as channel:
+        stub = pay_grpc_client.KodyPayTerminalServiceStub(channel)
+        response = stub.Terminals(
+            pay_model.TerminalsRequest(store_id=config.store_id),
+            metadata=[("x-api-key", config.api_key)]
+        )
+````
+- TerminalResponse : Terminal Response
+
+```python
+message TerminalsResponse {
+  repeated Terminal terminals = 1;
+}
+
+message Terminal {
+  string terminal_id = 1; // terminal serial number
+  bool online = 2;
+}
 ```
 
+### **Create terminal payment**
+
+Send to a terminal a payment initiation request. This request can make the terminal to immediately display a card acquiring screen, or display a tip screen to the user which after will go to the card scanning screen.
+
+````python
+# without tips screen
+with grpc.secure_channel(target=config.address, credentials=grpc.ssl_channel_credentials()) as channel:
+        stub = pay_grpc_client.KodyPayTerminalServiceStub(channel)
+        response_iterator = stub.Pay(
+            pay_model.PayRequest(store_id=config.store_id, terminal_id=config.terminal_id, amount=amount),
+            metadata=[("x-api-key", config.api_key)]
+        )
+
+# with tips screen
+with grpc.secure_channel(target=config.address, credentials=grpc.ssl_channel_credentials()) as channel:
+        stub = pay_grpc_client.KodyPayTerminalServiceStub(channel)
+        response_iterator = stub.Pay(
+            pay_model.PayRequest(store_id=config.store_id, terminal_id=config.terminal_id, amount=amount,  show_tips=bool(True)),
+            metadata=[("x-api-key", config.api_key)]
+        )
+````
+
+- PayResponse : Payment Response
+
+````python
+message PayResponse {
+  PaymentStatus status = 1;
+  optional string failure_reason = 2; // only populated on failure
+  optional string receipt_json = 3; // json blob containing the receipt data
+  string order_id = 4;
+  google.protobuf.Timestamp date_created = 5;
+  optional string ext_payment_ref = 6;
+  google.protobuf.Timestamp date_paid = 7;
+  optional string total_amount = 8;
+  optional string sale_amount = 9;
+  optional string tips_amount = 10;
+}
+````
+
+### **Get Terminal Payment Details**
+
+````python
+with grpc.secure_channel(target=config.address, credentials=grpc.ssl_channel_credentials()) as channel:
+        stub = pay_grpc_client.KodyPayTerminalServiceStub(channel)
+        response = stub.PaymentDetails(
+            pay_model.PaymentDetailsRequest(store_id=config.store_id, order_id=config.order_id),
+            metadata=[("x-api-key", config.api_key)]
+        )
+````
+- PayResponse : Get Payment Detail Response
+
+### **Cancel Terminal Payment**
+
```

Comment:
The cancel payment request requires the following parameters:
- `store_id` - the ID of your assigned store
- `terminal_id` - the serial number of the terminal that is processing the payment request
- `amount` - the amount sent in the original payment request, used to find the payment request
- `order_id` - the Order ID returned in the initial payment response, a unique UUID value for each payment

The request authenticates with the server using the `x-api-key` metadata field, you must use your assigned API key.